crates/nu-cli/src/repl.rs

```diff
@@ -140,6 +141,16 @@ pub fn evaluate_repl(
         }
     }
 
+    if let Some(s) = prerun_command {
+        eval_source(
+            engine_state,
+            stack,
+            s.item.as_bytes(),
+            &format!("entry #{}", entry_num),
+            PipelineData::new(Span::new(0, 0)),
+        );
```

Comment:
You'll need to add `engine_state.merge_env()` here, otherwise, some things, like changing a directory, won't be seenby the parser which would impact commands like `source`, `use`, or `register.